torch/distributed/checkpoint/examples/stateful_example.py

```diff
@@ -0,0 +1,104 @@
+# Owner(s): ["oncall: distributed"]
+
+
+import os
+import shutil
+
+import torch
+import torch.distributed as dist
+import torch.distributed.checkpoint as DCP
+import torch.multiprocessing as mp
+import torch.nn as nn
+from torch.distributed._tensor.device_mesh import init_device_mesh
+from torch.distributed.checkpoint.state_dict import (
+    _patch_model_state_dict,
+    _patch_optimizer_state_dict,
+)
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+
+
+CHECKPOINT_DIR = f"~/{os.environ['LOGNAME']}/checkpoint"
+
+
+class Model(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        torch.manual_seed(0)
+        self.net1 = nn.Sequential(nn.Linear(8, 16), nn.ReLU())
+        self.net2 = nn.Sequential(nn.Linear(16, 32), nn.ReLU())
+        self.net3 = nn.Linear(32, 64)
```

Comment:
nit, we should add ReLU

Reply:
@fegin could you clarify where the suggested Relu would go?

Reply:
Just the same as `net2`.